torchtune/datasets/_packed.py

```diff
@@ -209,14 +208,12 @@ def _add_pack(self, pack: PACK_TYPE) -> None:
         self.packs.append(pack)
 
     def _convert_to_tensors(self, pack: PACK_TYPE) -> PACK_TYPE:
-        """Converts a pack into tensors. Pack comes in as a dict of lists and is converted to tensors.
-        The only key that does not get converted is ``seq_lens``.
-        """
+        """Converts a pack into tensors. Pack comes in as a dict of lists and is converted to tensors."""
         return {
-            "tokens": torch.tensor(pack["tokens"]),
-            "labels": torch.tensor(pack["labels"]),
-            "input_pos": torch.tensor(pack["input_pos"]),
-            "seq_lens": pack["seq_lens"],
+            "tokens": torch.tensor(pack["tokens"], dtype=torch.long),
+            "labels": torch.tensor(pack["labels"], dtype=torch.long),
+            "input_pos": torch.tensor(pack["input_pos"], dtype=torch.long),
+            "seq_lens": torch.tensor(pack["seq_lens"], dtype=torch.long),
```

Comment:
I thought the point of `PACK_TYPE` was to let this be a List. Isn't it just `Dict[str, Tensor]` now?

Reply:
the input is still a list. we need to convert to tensors so we can pad the individual packs independent of collater.